beaker/client/logic_error.py

```diff
@@ -38,6 +38,8 @@ def __str__(self) -> str:
         return f"Txn {self.txid} had error '{self.msg}' at PC {self.pc} and Source Line {self.line_no}: \n\n\t{self.trace()}"
 
     def trace(self, lines: int = 5) -> str:
+        if self.line_no is None:
```

Comment:
Ah so, I was going to suggest a bit of a refactor here anyway - I _think_ this class should only be used if the `LOGIC_ERROR` regex does match. There is a psuedo-check by testing if `logic in str(e)` each time before `wrap_approval_exception(e)` is called, which is the only place `LogicException` is used - I would suggest removing that repeated psuedo-check in each call, checking for the regex match inside wrap_approval_exception, and only constructing a `LogicException` if it matches. I will submit a PR for that today I hope.

Reply:
I can do this unless you've got something started?

Reply:
Oops sorry just saw this now, pushed up PR #158

Reply:
looks great! thanks!